tcms/issuetracker/bitbucket.py

```diff
@@ -0,0 +1,146 @@
+        self.headers = {"Accept": 'application/json',
+                        'Content-type': 'application/json'}
+        self.auth = HTTPBasicAuth(api_username, api_password)
+
+    def create_issue(self, data):
+        url = '{0}{1}'.format(self.endpoint_url, '/issues')
+        return self._request("POST", url, headers=self.headers, auth=self.auth, json=data)
+
+    def get_issue(self, issue_id):
+        url = '{0}{1}{2}'.format(self.endpoint_url, '/issues/', issue_id)
+        return self._request("GET", url, headers=self.headers, auth=self.auth)
+
+    def update_issue(self, issue_id, data):
+        url = '{0}{1}{2}{3}'.format(self.endpoint_url, '/issues/', issue_id, '/changes')
+        return self._request("POST", url, headers=self.headers, auth=self.auth, json=data)
+
+    def add_comment(self, issue_id, comment):
+        url = '{0}{1}{2}{3}'.format(self.endpoint_url, '/issues/', issue_id, '/comments/')
+        return self._request("POST", url, headers=self.headers, auth=self.auth, json=comment)
+
+    def get_comments(self, issue_id):
+        url = '{0}{1}{2}{3}'.format(self.endpoint_url, '/issues/',
+                                    issue_id, '/comments?sort=-updated_on')
+        return self._request("GET", url, headers=self.headers, auth=self.auth)
+
+    @staticmethod
+    def _request(method, url, **kwargs):
+        return requests.request(method, url, **kwargs).json()
+
+    @staticmethod
+    def _construct_endpoint_url(api_version, url):
+        splitted_url = url.replace('https://', '').split('/')
+        base_url = "https://api.bitbucket.org"
+        workspace = splitted_url[1]
+        repository = splitted_url[2]
+        endpoint_url = "{0}/{1}/{2}/{3}/{4}".format(base_url,
+                                                    api_version,
+                                                    'repositories',
+                                                    workspace,
+                                                    repository)
+        return endpoint_url
+
+
+class BitBucketThread(IntegrationThread):
+    """
+        Execute BitBucket code in a thread!
+
+        Executed from the IssueTracker interface methods.
+    """
+
+    def post_comment(self):
+        comment_body = {"content": {"raw": self.text().replace('\n', '\n\n')}}
+        self.rpc.add_comment(self.bug_id, comment_body)
+
+
+class BitBucket(IssueTrackerType):
+    """
+        Support for BitBucket. Requires:
+
+        :base_url: - e.g. https://bitbucket.org/{workspace}/{repository}
+        :api_username: BitBucket Username Required
+        :api_password: BitBucket App Password Required
+
+        .. note::
+
+            You can leave the ``api_url`` field blank because
+            the integration code doesn't use them!
+
+            ``api_username`` is your BitBucket username, which you have
+            created the ``api_password`` and it is required.
```

Comment:
`required` - redundant. Here you need to point out this is the username used to login via the BB website.
Also good to add link to https://support.atlassian.com/bitbucket-cloud/docs/app-passwords/  and another one for api_password that leads to instructions how to create the "App password" token.